<?xml version="1.0"?>
<launch>

  <!-- send cob3 urdf to param server -->
  <include file="$(find cob_hardware_config)/common/upload_robot.launch" />

  <!-- push robot_description to factory and spawn robot in gazebo -->
  <node name="spawn_gazebo_model" pkg="gazebo" type="spawn_model" args="-urdf -param robot_description -model robot -z 0.1 " respawn="false" output="screen" />

  <node name="create_colored_point_cloud" pkg="cob_gazebo" type="create_colored_point_cloud" respawn="false" output="screen">
    <remap from="image_color" to="cam3d/rgb/image_raw"/>
    <remap from="point_cloud2" to="cam3d/depth/points"/>
    <remap from="colored_point_cloud2" to="cam3d/rgb/points"/>
  </node>
  <!-- convert from PointCloud2 to PointCloud1 -->
  <node name="point_cloud_converter_full_cloud" pkg="cob_gazebo" type="point_cloud_converter" args="points2_in:=cam3d/depth/points points_out:=cam3d/depth/points1" respawn="false" output="screen" />

</launch>
